src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -16,10 +16,25 @@
 
 namespace Microsoft.Data.SqlClient
 {
+    internal readonly ref struct SessionHandle
+    {
+        public readonly SNIHandle NativeHandle;
+
+        public SessionHandle(SNIHandle nativeHandle) => NativeHandle = nativeHandle;
+        
+        public bool IsNull => NativeHandle is null;
+    }
+
     internal partial class TdsParserStateObject
     {
+        private static bool UseManagedSNI => false;
+
         private SNIHandle _sessionHandle = null;              // the SNI handle we're to work on
 
+        private SessionHandle SessionHandle => new SessionHandle(_sessionHandle);
+
+        private bool TransparentNetworkIPResolution => _parser.Connection.ConnectionOptions.TransparentNetworkIPResolution;
+
         internal bool _pendingData = false;
         internal bool _errorTokenReceived = false;               // Keep track of whether an error was received for the result.
```

Comment:
Could you make private this field and add the same property with get/set on it  as NetCore:
```csharp
        internal bool HasReceivedError
        {
            get => _errorTokenReceived;
            set => _errorTokenReceived= value;
        }
```